packages/components/select/src/option.vue

```diff
@@ -67,7 +67,15 @@ export default defineComponent({
     const vm = getCurrentInstance().proxy
     const key = (vm as unknown as SelectOptionProxy).value
     select.onOptionCreate(vm as unknown as SelectOptionProxy)
-
+    // fix: 8544
+    //  When props are changed, the component is remounted,
+    //  and "onOptionCreate" will be called,
+    //  but in the previous hook method "onBeforeUnmount",
+    // "cachedOptions.delete" will be executed after "onOptionCreate", which leads to a bug (8544).
+    // Here, call "onOptionCreate" again
+    nextTick(() => {
+      select.onOptionCreate(vm as unknown as SelectOptionProxy)
+    })
```

Comment:
I do not think this is the best way, too many nextTick will make the component difficult to maintain.

I think the problem is in line 78, the if should be inside nextTick.